pandas/tests/groupby/test_transform.py

```diff
@@ -722,19 +722,29 @@ def interweave(list_obj):
 
 
 @pytest.mark.parametrize("test_series", [True, False])
+@pytest.mark.parametrize("shuffle", [True, False])
 @pytest.mark.parametrize("periods,fill_method,limit", [
     (1, 'ffill', None), (1, 'ffill', 1),
     (1, 'bfill', None), (1, 'bfill', 1),
     (-1, 'ffill', None), (-1, 'ffill', 1),
     (-1, 'bfill', None), (-1, 'bfill', 1)])
-def test_pct_change(test_series, periods, fill_method, limit):
-    vals = [np.nan, np.nan, 1, 2, 4, 10, np.nan, np.nan]
-    exp_vals = Series(vals).pct_change(periods=periods,
-                                       fill_method=fill_method,
-                                       limit=limit).tolist()
-
-    df = DataFrame({'key': ['a'] * len(vals) + ['b'] * len(vals),
-                    'vals': vals * 2})
+def test_pct_change(test_series, shuffle, periods, fill_method, limit):
+    vals = [3, np.nan, 1, 2, 4, 10, np.nan, np.nan]
+    keys = ['a', 'b']
+    key_v = [k for j in list(map(lambda x: [x] * len(vals), keys)) for k in j]
+    df = DataFrame({'key': key_v, 'vals': vals * 2})
+    if shuffle:
+        order = np.random.RandomState(seed=42).permutation(len(df))
+        df = df.reindex(order).reset_index(drop=True)
+
+    manual_apply = []
+    for k in keys:
```

Comment:
why aren't you just using this? ``df.loc[df.key == k, 'vals']``, which is a Series